### ISSUE TITLE
No GPU device

### ISSUE THREAD

**Issue by boulevard-h** [openwall/john#5262]
My OS is Windows11 22H2, and I use CygWin to build JtR following this instruction: [https://github.com/openwall/john/blob/bleeding-jumbo/doc/INSTALL-WINDOWS](url)

I successfully build John the Ripper and run the `john --test=0`
However, I cannot find my GPU(RTX 2070 mobile):
```
boulevard@boulevard-TX9 ~/john-bleeding-jumbo/john-bleeding-jumbo/src
$ ../run/john --list=opencl-devices
Platform #0 name: Portable Computing Language, version: OpenCL 1.2 pocl 1.3 RelWithDebInfo+Asserts, LLVM 8.0.1, SPIR, SLEEF, POCL_DEBUG
    Device #0 (1) name:     pthread-Intel(R) Core(TM) i5-10400 CPU @ 2.90GHz
    Device vendor:          GenuineIntel
    Device type:            CPU (LE)
    Device version:         OpenCL 1.2 pocl HSTR: pthread-x86_64-unknown-windows-cygnus-bdver2
    OpenCL version support: OpenCL C 1.2 pocl
    Driver version:         1.3
    Native vector widths:   char 16, short 16, int 8, long 4
    Preferred vector width: char 16, short 16, int 8, long 4
    Global Memory:          7352 MiB
    Global Memory Cache:    12 MiB
    Local Memory:           8 MiB (Global)
    Constant Buffer size:   8 MiB
    Max memory alloc. size: 2 GiB
    Max clock (MHz):        2910
    Profiling timer res.:   1 ns
    Max Work Group Size:    4096
    Parallel compute cores: 12
    Speed index:            139680
```
I have the nvidia-driver and cuda on my computer:
```
boulevard@boulevard-TX9 ~/john-bleeding-jumbo/john-bleeding-jumbo/src
$ nvidia-smi
Thu Mar 23 16:55:25 2023
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 528.49       Driver Version: 528.49       CUDA Version: 12.0     |
|-------------------------------+----------------------+----------------------+
| GPU  Name            TCC/WDDM | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA GeForce ... WDDM  | 00000000:01:00.0 Off |                  N/A |
| N/A   48C    P8     8W / 115W |    183MiB /  8192MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A      1756    C+G   C:\Windows\System32\dwm.exe     N/A      |
|    0   N/A  N/A      5344    C+G   ...2txyewy\TextInputHost.exe    N/A      |
|    0   N/A  N/A      6224    C+G   ...ysdiag\bin\HipsDaemon.exe    N/A      |
|    0   N/A  N/A      9656    C+G   ...n1h2txyewy\SearchHost.exe    N/A      |
|    0   N/A  N/A      9696    C+G   ...artMenuExperienceHost.exe    N/A      |
|    0   N/A  N/A     11772    C+G   ...e\Current\LogiOverlay.exe    N/A      |
|    0   N/A  N/A     16368    C+G   ...661.44\msedgewebview2.exe    N/A      |
|    0   N/A  N/A     17112    C+G   ...kle-tray\Twinkle Tray.exe    N/A      |
+-----------------------------------------------------------------------------+
```

What can I do to use my GPU, appreciate for your help.

**Comment by solardiz** [comment]
This is probably an instance of #4327, so please try the workaround from there. Longer-term, we should fix #2133.

Mid-term, we could want to update our documentation (perhaps `README-OPENCL` and `INSTALL-WINDOWS`) to include the workaround. @claudioandre-br What do you think? If you agree, then please create a pull request with such edits. Thanks!

**Comment by boulevard-h** [comment]
> This is probably an instance of #4327, so please try the workaround from there. Longer-term, we should fix #2133.
> 
> Mid-term, we could want to update our documentation (perhaps `README-OPENCL` and `INSTALL-WINDOWS`) to include the workaround. @claudioandre-br What do you think? If you agree, then please create a pull request with such edits. Thanks!

I find one way to solve it:
1. Find `cygOpenCL.dll` in `C:\cygwin64\bin\`, rename it as `cygOpenCL-1.dll.bac`
2. Copy `C:\Windows\System32\OpenCL.dll` to  `C:\cygwin64\bin\`, and rename it as `cygOpenCL.dll`
Then I can find my GPU

**Comment by solardiz** [comment]
Thank you, @boulevard-h. I've just added a related comment to #3191 and suggest that we use that issue for discussion of a possible fix. I'll close this one.
